clar2wasm/src/words/options.rs

```diff
@@ -0,0 +1,89 @@
+use crate::wasm_generator::{drop_value, ArgumentsExt, GeneratorError, WasmGenerator};
+use clarity::vm::{types::TypeSignature, ClarityName, SymbolicExpression};
+use walrus::ir::BinaryOp;
+
+use super::Word;
+
+#[derive(Debug)]
+pub struct IsSome;
+
+impl Word for IsSome {
+    fn name(&self) -> ClarityName {
+        "is-some".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let opt = args.get_expr(0)?;
+        generator.traverse_expr(builder, opt)?;
+        // there is an optional type on top of the stack.
+
+        // Get the type of the optional expression
+        let ty = generator
+            .get_expr_type(opt)
+            .expect("input expression must be typed")
+            .clone();
+
+        let some_ty = if let TypeSignature::OptionalType(some_type) = &ty {
+            &**some_type
+        } else {
+            panic!("Expected an Optional type. Found: {:?}", ty);
+        };
```

Comment:
Maybe this part, which is common for both functions, could be extracted into its own function?

Reply:
Added a function with common behaviour.